packages/website/blog/2022-09-03-automated-rule-docs-with-docusaurus-and-remark.md

```diff
@@ -0,0 +1,224 @@
+---
+authors:
+  - image_url: https://www.joshuakgoldberg.com/img/josh.jpg
+    name: Josh Goldberg
+    title: TypeScript ESLint Maintainer
+    url: https://github.com/JoshuaKGoldberg
+description: How TypeScript ESLint generates much of the docs pages for each of its lint rules.
+slug: automated-rule-docs-with-docusaurus-and-remark
+tags: [documentation, docusaurus, remark]
+title: Automated Rule Docs With Docusaurus and Remark
+---
+
+The TypeScript ESLint website at https://typescript-eslint.io is the canonical location for documentation on how to use ESLint on TypeScript code.
+The site includes a documentation page for each of the over 100 ESLint rules exposed by its ESLint plugin.
+Each of those rule docs pages includes a description of the rule, any options it allows, links to its source code, and other important information about its usage.
+
+Until recently, keeping descriptions of rules consistent between their source code and docs pages was a cumbersome manual chore.
+We'd written a suite of Jest tests to verify they matched -- but those tests didn't capture everything, often failed with cryptic messages, and were missed by less experienced contributors.
+
+We're happy to say that now, we've overhauled rule docs pages to automatically generate metadata information from rule source code.
+That means the pages always display up-to-date information without developers needing to manually rewrite docs on rule changes.
+Hooray! 🎉
+
+This blog post gives an overview of the [chore(website): generate rule docs options automatically](https://github.com/typescript-eslint/typescript-eslint/pull/5386).
+
+<!--truncate-->
+
+## Tools In Play
+
+https://typescript-eslint.io is built on [Docusaurus](https://docusaurus.io), a powerful static site generator tailored to documentation websites and blogs.
+Docusaurus comes with sensible defaults, extensive configuration options, and thoroughly fleshed out documentation.
+We're big fans.
+❤️
+
+[Docusaurus blogs](https://docusaurus.io/docs/blog) support writing blog posts in [MDX](https://mdxjs.com), a rich superset of Markdown that allows inserting JSX components.
+Docusaurus supports MDX using [remark](https://remark.js.org), a Markdown processor powered by plugins.
+Remark plugins take in Markdown files in a format known as an Abstract Syntax Trees, or AST, and output modified versions of those ASTs.
+
+You can see what Remark's AST equivalent of a Markdown document looks like by visiting [AST Explorer](https://astexplorer.net) and selecting the Markdown language.
+
+![astexplorer.net showing the default snippet with the paragraph highlighted](./ast-explorer-remark.png)
+
+> _astexplorer.net showing the default snippet with the paragraph highlighted_
```

Comment:
I suggest using `<figure>` and `<figcaption>`. The blockquote is honestly confusing from a typesetting perspective when I was reading it. (Also, alt messages should usually be different from the caption, so this one is good)

Reply:
Aha! I was wondering if there was a better way to do this. Thank you!